sql/hive/src/main/java/org/apache/hadoop/hive/ql/exec/SparkDefaultUDFMethodResolver.java

```diff
@@ -0,0 +1,341 @@
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.exec;
+
+import org.apache.hadoop.hive.serde2.objectinspector.PrimitiveObjectInspector.PrimitiveCategory;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorUtils.PrimitiveGrouping;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector.Category;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorUtils;
+import org.apache.hadoop.hive.serde2.typeinfo.*;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.spark.internal.SparkLogger;
+import org.apache.spark.internal.SparkLoggerFactory;
+
+/**
+ * A equivalent implementation of {@link DefaultUDFMethodResolver}, but eliminate calls
+ * of {@link org.apache.hadoop.hive.ql.exec.FunctionRegistry} to avoid initializing Hive
+ * built-in UDFs.
+ */
+public class SparkDefaultUDFMethodResolver implements UDFMethodResolver {
+
+  public static final SparkLogger LOG =
+          SparkLoggerFactory.getLogger(SparkDefaultUDFMethodResolver.class);
+
+  /**
+   * The class of the UDF.
+   */
+  private final Class<? extends UDF> udfClass;
+
+  /**
+   * Constructor. This constructor extract udfClass from {@link DefaultUDFMethodResolver}
+   */
+  @SuppressWarnings("unchecked")
+  public SparkDefaultUDFMethodResolver(
+      DefaultUDFMethodResolver wrapped) throws ReflectiveOperationException {
+    Field udfClassField = wrapped.getClass().getDeclaredField("udfClass");
+    udfClassField.setAccessible(true);
+    this.udfClass = (Class<? extends UDF>) udfClassField.get(wrapped);
+  }
+
+  /**
+   * Gets the evaluate method for the UDF given the parameter types.
+   *
+   * @param argClasses
+   *          The list of the argument types that need to matched with the
+   *          evaluate function signature.
+   */
+  @Override
+  public Method getEvalMethod(List<TypeInfo> argClasses) throws UDFArgumentException {
+    return getMethodInternal(udfClass, "evaluate", false, argClasses);
+  }
+
+  // Below methods are copied from Hive 2.3.10 o.a.h.hive.ql.exec.FunctionRegistry
```

Comment:
above is wrapper code, below is copied code.

Reply:
As discussed offline, if we proceed with the current pr, we need to refer to the description in https://github.com/apache/spark/pull/49736 to ensure that the hive-thriftserver tests can be successfully executed after removing the hive-llap dependency.

Reply:
Yes, I wasn't aware of the STS issue previously, this approach requires further investigation.

Reply:
I believe the STS issue is addressed now. @LuciferYang could you please take another look?

```
$ build/mvn -Phive-thriftserver install -DskipTests
$ build/mvn -pl sql/hive-thriftserver -Phive-thriftserver install -fae
...
Run completed in 17 minutes, 21 seconds.
Total number of tests run: 639
Suites: completed 20, aborted 0
Tests: succeeded 639, failed 0, canceled 0, ignored 25, pending 0
All tests passed.
```